File: reservation-frontend/src/app/models/reservation.ts
import { Guid } from "guid-typescript"
import { Ticket } from "./ticket"

export class Reservation {
    public id!: number
    public userId!: Guid
    public name!: string
    public email!: string
    public phoneNumber!: string
    public tickets!: Ticket[]
    public eventOccurrenceId!: number
    public paymentCompleted!: boolean
    public isDeleted!: boolean
}

File: reservations-api/src/Reservations.Domain/Reservation.cs
﻿namespace Reservations.Domain
{
    public class Reservation : AuditableEntity
    {
        public Reservation()
        {
            Tickets = new HashSet<Ticket>();
        }
        public long Id { get; set; }
        public Guid UserId { get; set; }
        public User User { get; set; }
        public ICollection<Ticket> Tickets { get; set; }
        public long EventOccurrenceId { get; set; }
        public EventOccurrence EventOccurrence { get; set; }
        public bool PaymentCompleted { get; set; }
    }
}